site/bulletin_board/routes/users.js

```diff
@@ -84,4 +86,26 @@ router.post('/', (req, res, next) => {
   });
 });
 
+router.get('/profile', (req, res, next) => {
+  res.redirect("profile/"+req.session.passport.user);
```

Comment:
Clever! But you might consider the performance implications of this. One possible alternative:

```
let get_profile_data = ( profile_user_id, viewer_user_id, callback ) => {
  datasource.retrieve(profile_user_id, viewer_user_id, (user) => {
    postsourece.retrieveperuser(profile_user_id, (posts) => {
      postsourece.post_liked(profile_user_id, (posts_liked) => {
         callback({ name : user.username, birthdate : user.birthdate , user: user ,posts : posts,posts_liked:posts_liked});
      });
    });
  });
}; 

router.get('/profile', (req, res, next) => {
  get_profile_data((req.user.id, req.user.id, data) => { res.render('profile_view', data); });
});

router.get('/profile/:id', (req, res, next) => {
  get_profile_data((req.params['id'], req.user.id, data) => { res.render('profile_view', data); });
});

router.get('/:id', (req, res, next) => {
  get_profile_data((req.params['id'], req.user.id, data) => { res.render('user_view', data); });
});
```